verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -402,10 +426,32 @@ void IndexingFactsTreeExtractor::ExtractModulePort(
     const SyntaxTreeLeaf* leaf =
         GetIdentifierFromPortReference(module_port_node);
 
-    facts_tree_context_.top().NewChild(IndexingNodeData(
-        {Anchor(leaf->get(), context_.base)},
-        has_propagated_type ? IndexingFactType::kVariableDefinition
-                            : IndexingFactType::kVariableReference));
+    if (has_propagated_type) {
```

Comment:
Thought: instead of a `bool`, what if you passed in a maybe-null pointer to propagated type information?  Between calls to handling port declarations, that type would be updated.  When a port has an explicit type, you'd just ignore the previous propagated type.

(Could be follow-up, if this change is too invasive now.)

Reply:
Maybe in the upcoming refactoring. I am thinking about doing some refactoring anyway.